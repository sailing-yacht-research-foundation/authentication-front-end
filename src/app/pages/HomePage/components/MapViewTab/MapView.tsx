import 'leaflet/dist/leaflet.css';

import React, { useEffect, useImperativeHandle } from 'react';
import { useMap } from 'react-leaflet';
import * as L from 'leaflet';
import ReactDOMServer from 'react-dom/server';
import { GiSailboat } from 'react-icons/gi';

const races = [
    {
        latlng: {
            lng: -343.167625,
            lat: 38.001357
        },
        name: 'Race over here'
    },
    {
        latlng: {
            lng: -346.729520,
            lat: 40.885278
        },
        name: 'Race over here'
    },
    {
        latlng: {
            lng: -355.527362,
            lat: 41.944783
        },
        name: 'Race over here'
    },
    {
        latlng: {
            lng: -368.117287,
            lat: 51.380010
        },
        name: 'Race over here'
    }
];

const MAP_MOVE_TYPE = {
    immediately: 'immediately',
    animation: 'animation'
}

export const MapView = React.forwardRef<any, any>(({ zoom }, ref) => {

    const map = useMap();

    useEffect(() => {
        initializeMapView();
        zoomToCurrentUserLocationIfAllowed(MAP_MOVE_TYPE.immediately);
        attachRaceMarkersToMap();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    useImperativeHandle(ref, () => ({
        zoomToCurrentUserLocationIfAllowed() {
            zoomToCurrentUserLocationIfAllowed(MAP_MOVE_TYPE.animation);
        }
      }));

    /**
     * Zoom to the location of current user
     * @param type should be either type: immediately or animation
     */
    const zoomToCurrentUserLocationIfAllowed = (type: string) => {
        if (navigator.geolocation)
            navigator.geolocation.getCurrentPosition((position) => {
                const params = {
                    lat: position.coords.latitude,
                    lng: position.coords.longitude
                };
                
                type === MAP_MOVE_TYPE.animation ? map.flyTo(params, zoom) : map.setView(params, zoom);
            });
    }

    const initializeMapView = () => {
        new L.TileLayer(`https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=${process.env.REACT_APP_MAP_BOX_API_KEY}`, {
            attribution: '<a href="https://www.github.com/sailing-yacht-research-foundation"><img src="https://syrf.io/wp-content/themes/syrf/assets/svg/icon-github.svg"></img></a>',
            maxZoom: 18,
            id: 'jweisbaum89/cki2dpc9a2s7919o8jqyh1gss',
            tileSize: 512,
            zoomOffset: -1,
            accessToken: 'your.mapbox.access.token'
        }).addTo(map);
    }

    const attachRaceMarkersToMap = () => {
        races.forEach(race => {
            L.marker(race.latlng, {
                icon: L.divIcon({
                    html: ReactDOMServer.renderToString(<GiSailboat style={{ color: '#fff', fontSize: '35px' }} />),
                    iconSize: [20, 20],
                    className: 'myDivIcon'
                })
            })
            .addTo(map);
        });
    }

    return (
        <>
        </>
    );
})