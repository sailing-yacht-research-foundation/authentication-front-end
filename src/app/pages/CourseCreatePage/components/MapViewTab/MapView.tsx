import 'leaflet/dist/leaflet.css';
import "leaflet-draw/dist/leaflet.draw-src.css";

import React from 'react';
import { useMap } from 'react-leaflet';
import * as L from 'leaflet';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { selectIsAuthenticated } from 'app/pages/LoginPage/slice/selectors';
import { v4 as uuidv4 } from 'uuid';
import { Modal, Form } from 'antd';
import { SyrfFieldLabel, SyrfInputField } from 'app/components/SyrfForm';
import { useCourseSlice } from '../../slice';
import { selectCourseSequencedGeometries } from '../../slice/selectors';
import styled from 'styled-components';
import { FaSave } from 'react-icons/fa';
import { toast } from 'react-toastify';
import { useHistory } from 'react-router-dom';

require('leaflet-draw');

type CourseGeometry = {
    geometryType: string;
    coordinates: any[][];
    properties: any;
    order: number;
    id: string;
};

const LAYER_TYPE = {
    marker: 'marker',
    polygon: 'polygon',
    polyline: 'polyline'
};

const GEOMETRY_TYPE = {
    point: 'Point',
    line: 'Polyline',
    polygon: 'Polygon'
};

export const MapView = () => {

    const map = useMap();

    const isAuthenticated = useSelector(selectIsAuthenticated);

    const layerOrder = React.useRef(1);

    const geometryName = React.useRef('');

    const [showGeometryNamePopup, setShowGeometryNamePopup] = React.useState<boolean>(false);

    const [geometryNameForm] = Form.useForm();

    const { actions } = useCourseSlice();

    const couseSequencedGeometries = useSelector(selectCourseSequencedGeometries);

    const mutableCouseSequencedGeometries = React.useRef<any[]>([]);

    const history = useHistory();

    const dispatch = useDispatch();

    const initializeMapView = () => {
        new L.TileLayer(`https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=${process.env.REACT_APP_MAP_BOX_API_KEY}`, {
            attribution: '<a href="https://www.github.com/sailing-yacht-research-foundation"><img src="https://syrf.io/wp-content/themes/syrf/assets/svg/icon-github.svg"></img></a>',
            maxZoom: 18,
            minZoom: 2,
            id: 'jweisbaum89/cki2dpc9a2s7919o8jqyh1gss',
            tileSize: 512,
            zoomOffset: -1,
            accessToken: 'your.mapbox.access.token'
        }).addTo(map);

        let drawnItems = L.featureGroup().addTo(map);
        map.addControl(new L.Control.Draw({
            edit: {
                featureGroup: drawnItems,
                poly: {
                    allowIntersection: false
                }
            },
            draw: {
                rectangle: false,
                circle: false,
                circlemarker: false,
                polygon: {
                    allowIntersection: false,
                    showArea: true
                }
            }
        }));

        return drawnItems;
    }

    useEffect(() => {
        let drawnItems = initializeMapView();
        registerOnGeometryDrawStartEvent();
        registerOnGeometryCreatedEvent(drawnItems);
        registerOnSavedEvent();
        registerOnLayerDeletedEvent();
    }, []);

    const registerOnSavedEvent = () => {
        map.on(L.Draw.Event.EDITSTOP, function (e) {

        });
    }

    const registerOnLayerDeletedEvent = () => {
        map.on(L.Draw.Event.DELETED, function (e) {
            e.layers.eachLayer(layer => {
                mutableCouseSequencedGeometries.current.filter(geometry => {
                    return geometry.id !== layer.options._id;
                })
            });
            dispatch(actions.setCourseSequencedGeometries(mutableCouseSequencedGeometries.current));
        });
    }

    const registerOnGeometryDrawStartEvent = () => {
        map.on(L.Draw.Event.DRAWSTART, function (e) {
            setShowGeometryNamePopup(true);
        });
    }

    const registerOnGeometryCreatedEvent = (drawnItems) => {
        map.on(L.Draw.Event.CREATED, function (e) {
            let layerType = e.layerType,
                layer = e.layer,
                layerId = uuidv4(),
                geometry: CourseGeometry | any = {
                    id: layerId,
                    properties: {
                        name: geometryName.current
                    },
                    order: layerOrder.current,
                };

            switch (layerType) {
                case LAYER_TYPE.marker:
                    geometry.geometryType = GEOMETRY_TYPE.point;
                    geometry.coordinates = [
                        [layer.getLatLng().lat, layer.getLatLng().lng]
                    ];
                    layer.options._geometry_type = GEOMETRY_TYPE.point;
                    break;
                case LAYER_TYPE.polygon:
                    geometry.geometryType = GEOMETRY_TYPE.polygon;
                    geometry.coordinates = [layer.getLatLngs().map(function (point) {
                        return [point.lat, point.lng];
                    })];
                    layer.options._geometry_type = GEOMETRY_TYPE.polygon;
                    break;
                case LAYER_TYPE.polyline:
                    geometry.geometryType = GEOMETRY_TYPE.line;
                    geometry.coordinates = [layer.getLatLngs().map(function (point) {
                        return [point.lat, point.lng];
                    })];
                    layer.options._geometry_type = GEOMETRY_TYPE.line;
                    break;
            }

            layer.options._name = geometryName.current;
            layer.options._id = layerId;
            drawnItems.addLayer(layer);
            mutableCouseSequencedGeometries.current.push(geometry);
            dispatch(actions.setCourseSequencedGeometries(JSON.parse(JSON.stringify(mutableCouseSequencedGeometries.current))));
            layerOrder.current++;
        });
    }

    const submitAndsetMarketName = () => {
        geometryNameForm
            .validateFields()
            .then(values => {
                const { geometry_name } = values;
                geometryName.current = geometry_name;
                setShowGeometryNamePopup(false);
            })
            .catch(info => {
            });
    }

    const saveCourse = () => {
        if (couseSequencedGeometries.length === 0) {
            toast.error('You cannot save the course because You have not added any geometries');
        }

        if (!isAuthenticated) {
            history.push('/signin?redirect_back=courses/create');
            toast.error('Please signup to continue!');
        }
    }

    return (
        <>
            <Modal
                title={'Enter geometry name'}
                bodyStyle={{ display: 'flex', justifyContent: 'center', overflow: 'hidden' }}
                cancelButtonProps={{ style: { display: 'none' } }}
                onOk={submitAndsetMarketName}
                closable={false}
                keyboard={false}
                visible={showGeometryNamePopup}>
                <Form
                    form={geometryNameForm}
                    layout="vertical"
                    name="basic"
                    style={{ width: '100%' }}

                    initialValues={{
                        geometry_name: '',
                    }}
                >
                    <Form.Item
                        label={<SyrfFieldLabel>Geometry name</SyrfFieldLabel>}
                        name="geometry_name"
                        rules={[{ required: true }]}
                    >
                        <SyrfInputField
                            placeholder={'Input a name for this geometry'}
                        />
                    </Form.Item>
                </Form>

            </Modal>
            <SaveButton onClick={saveCourse} title={'Save course'}>
                <FaSave style={{color: '#fff'}}/>
            </SaveButton>
        </>
    );
}

const SaveButton = styled.div`
    position: absolute;
    top: 290px;
    left: 12px;
    width: 30px;
    height: 30px;
    z-index: 9999;
    background: #599DF9;
    display: flex;
    justify-content: center;
    align-items:center;
    box-shadow: 0 1px 3px rgba(0,0,0,0.65);
    border-radius: 2px;
`;