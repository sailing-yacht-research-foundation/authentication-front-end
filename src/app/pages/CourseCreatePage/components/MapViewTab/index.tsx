import React from 'react';
import { MapContainer } from 'react-leaflet';
import styled from 'styled-components';
import { StyleConstants } from 'styles/StyleConstants';
import { MAP_DEFAULT_VALUE } from 'utils/helpers';
import { MapView } from './MapView';

export const MapViewTab = () => {

    return (
        <Wrapper>
            <MapContainer style={{ height: `calc(101vh - ${StyleConstants.NAV_BAR_HEIGHT} - ${StyleConstants.TAB_BAR_HEIGHT})`, width: 'calc(100%)' }} center={MAP_DEFAULT_VALUE.CENTER} zoom={MAP_DEFAULT_VALUE.ZOOM}>
                <MapView />
            </MapContainer>
        </Wrapper>
    );
}

const Wrapper = styled.div`
    display: flex;
`;