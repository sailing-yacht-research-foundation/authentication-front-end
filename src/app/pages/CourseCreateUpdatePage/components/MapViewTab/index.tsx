import { Button, Space } from 'antd';
import { DeleteButton, GobackButton, PageHeaderContainerResponsive, PageHeading, PageInfoContainer, PageInfoOutterWrapper } from 'app/components/SyrfGeneral';
import { translations } from 'locales/translations';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { BiSave, BiTrash } from 'react-icons/bi';
import { IoIosArrowBack } from 'react-icons/io';
import { MapContainer } from 'react-leaflet';
import { useLocation } from 'react-router';
import styled from 'styled-components';
import { StyleConstants } from 'styles/StyleConstants';
import { MAP_DEFAULT_VALUE } from 'utils/constants';
import { MapView } from './MapView';
import { MODE } from 'utils/constants';

const NAV_HEIGHT = '125px';

export const MapViewTab = () => {

    const { t } = useTranslation();

    const [mode, setMode] = React.useState<string>(MODE.CREATE);

    const mapViewRef = React.useRef<any>();

    const location = useLocation();

    React.useEffect(() => {
        if (location.pathname.includes(MODE.UPDATE))
            setMode(MODE.UPDATE);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <Wrapper>
            <PageHeaderContainerResponsive style={{ 'alignSelf': 'flex-start', width: '100%' }}>
                <PageInfoOutterWrapper>
                    <GobackButton onClick={() => mapViewRef.current.goBack()}>
                        <IoIosArrowBack style={{ fontSize: '40px', color: '#1890ff' }} />
                    </GobackButton>
                    <PageInfoContainer>
                        <PageHeading>{mode === MODE.UPDATE ? t(translations.course_create_update_page.update_course) : t(translations.course_create_update_page.create_a_new_course)}</PageHeading>
                    </PageInfoContainer>
                </PageInfoOutterWrapper>
                <Space size={10}>
                    <Button style={{ borderRadius: '5px' }} onClick={() => mapViewRef.current.saveCourse()} type="primary" icon={<BiSave style={{ marginRight: '5px' }} />}>
                        {t(translations.course_create_update_page.save)}
                    </Button>
                    {mode === MODE.UPDATE && <DeleteButton onClick={() => mapViewRef.current?.deleteCourse()} danger icon={<BiTrash
                        style={{ marginRight: '5px' }}
                        size={18} />}>{t(translations.course_create_update_page.delete)}</DeleteButton>}

                </Space>
            </PageHeaderContainerResponsive>
            <MapContainer style={{ height: `calc(101vh - ${StyleConstants.NAV_BAR_HEIGHT} - ${NAV_HEIGHT})`, width: 'calc(100%)' }} center={MAP_DEFAULT_VALUE.CENTER} zoom={MAP_DEFAULT_VALUE.ZOOM}>
                <MapView ref={mapViewRef} />
            </MapContainer>
        </Wrapper>
    );
}

const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
`;