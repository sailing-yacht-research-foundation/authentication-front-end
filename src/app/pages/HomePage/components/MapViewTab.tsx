import React from 'react';
import { StyleConstants } from 'styles/StyleConstants';
import { Input } from 'antd';
import styled from 'styled-components';
import { ReactComponent as SYRFLogo } from './assets/logo-dark.svg';
import { FilterPane } from '../FilterTab/components/FilterPane';
import { media } from 'styles/media';

interface StyledSearchBarProps {
    onClick: (e: Event) => void;
    onFocus: (e: Event) => void;
}

const TAB_BAR_HEIGHT = '76px';

export const MapViewTab = () => {

    const [showSearchPanel, setShowSearchPanel] = React.useState<boolean>(false);

    return (
        <>
            <iframe title="Google Maps"
                src="https://www.google.com/maps/embed?pb=!1m10!1m8!1m3!1d7862661.401754289!2d105.9102078!3d15.7939252!3m2!1i1024!2i768!4f13.1!5e0!3m2!1svi!2s!4v1627017281620!5m2!1svi!2s"
                style={{ width: '100%', height: `calc(100vh - ${StyleConstants.NAV_BAR_HEIGHT} - ${TAB_BAR_HEIGHT})` }} // 
                loading="lazy">
            </iframe>
            <SearchBarWrapper>
                <StyledSearchBar
                    onClick={e => setShowSearchPanel(true)}
                    onFocus={e => setShowSearchPanel(true)}
                    placeholder={'Search races with SYRF'} />
                <SearchBarLogo />
            </SearchBarWrapper>
            {showSearchPanel && <StyledSearchPane closable close={() => setShowSearchPanel(false)} />}
        </>
    )
}

const SearchBarWrapper = styled.div`
    position: absolute;
    top: 50%;
    left: 0;
    right: 0;
    margin: 0 auto;
    width: 530px;
    max-width: 80%;
`;

const SearchBarLogo = styled(SYRFLogo)`
    position: absolute;
    left: 15px;
    top: 0px;
    width: 45px;
    height: 45px;
`;

const StyledSearchBar = styled(Input) <StyledSearchBarProps>`
    border-radius: 10px;
    box-shadow: 0 3px 8px rgba(9, 32, 77, 0.12), 0 0 2px rgba(29, 17, 51, 0.12);
    padding-top: 10px;
    padding-bottom: 10px;
    text-indent: 60px;

    ::placeholder {
        font-weight: 500;
    }
`;

const StyledSearchPane = styled(FilterPane)`
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    margin: 0 auto;
    background: #fff;
    border-radius: 8px;
    padding: 0 20px;
    display: block;
    border-top: 1px solid #eee;
    width: 100%;

    ${media.medium`
        height: 550px;
        width: 530px;
    `}
`;