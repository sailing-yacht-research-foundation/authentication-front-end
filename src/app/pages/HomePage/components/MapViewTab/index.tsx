import 'leaflet/dist/leaflet.css';

import React, { useRef } from 'react';
import { Pagination } from 'antd';
import styled from 'styled-components';

import { media } from 'styles/media';
import { MapContainer, } from 'react-leaflet';
import { MapView } from './components/MapView';
import { translations } from 'locales/translations';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { selectPage, selectPageSize, selectResults, selectTotal } from '../../slice/selectors';
import { BiTargetLock } from 'react-icons/bi';
import { StyleConstants } from 'styles/StyleConstants';
import { selectUserCoordinate } from 'app/pages/LoginPage/slice/selectors';
import { MAP_DEFAULT_VALUE } from "utils/constants";

type MapViewProps = {
    zoomToCurrentUserLocationIfAllowed: () => void;
}

const ZOOM = 13;

export const MapViewTab = (props) => {

    const { onPaginationPageChanged } = props;

    const mapViewRef = useRef<MapViewProps>(null);

    const { t } = useTranslation();

    const total = useSelector(selectTotal);

    const page = useSelector(selectPage);

    const pageSize = useSelector(selectPageSize);

    const results = useSelector(selectResults);

    const userCoordinate = useSelector(selectUserCoordinate);

    const [isFocusingOnSearchInput,] = React.useState<boolean>(false);

    const mapContainerRef = React.useRef<any>();

    const zoomToUserLocation = () => {
        if (null !== mapViewRef.current) {
            mapViewRef.current.zoomToCurrentUserLocationIfAllowed();
        }
    }

    const handleResize = () => {
        const navHeight = 73;
        const tabHeight = 76;
        mapContainerRef.current._container.style.height = (window.innerHeight - navHeight - tabHeight) + 'px';
    }

    React.useEffect(() => {
        window.scroll(0, 0);
        window.addEventListener('resize', handleResize);
        return () => {
            window.removeEventListener('resize', handleResize);
        }
    }, []);

    const mapCenter = {
        lat: userCoordinate?.lat || MAP_DEFAULT_VALUE.CENTER.lat,
        lng: userCoordinate?.lon || MAP_DEFAULT_VALUE.CENTER.lng
    };

    return (
        <Wrapper>
            <MapContainer className="search-step-results" whenCreated={(mapInstance: any) => (mapContainerRef.current = mapInstance)} style={{ height: `calc(${window.innerHeight}px - ${StyleConstants.NAV_BAR_HEIGHT} - ${StyleConstants.TAB_BAR_HEIGHT})`, width: '100%', zIndex: 1 }} center={mapCenter} zoom={MAP_DEFAULT_VALUE.ZOOM}>
                <MapView isFocusingOnSearchInput={isFocusingOnSearchInput} ref={mapViewRef} zoom={ZOOM} />
            </MapContainer>
            {
                results.length > 0 && <PaginationWrapper>
                    <Pagination defaultCurrent={1} current={page} onChange={onPaginationPageChanged} total={total} pageSize={pageSize} />
                </PaginationWrapper>
            }
            {
                results.length === 0 && <MyLocationWrapper onClick={() => zoomToUserLocation()}>
                    <StyledMyLocationIcon />
                    <MyLocationText>{t(translations.home_page.map_view_tab.my_location)}</MyLocationText>
                </MyLocationWrapper>
            }
        </Wrapper>
    )
}

const Wrapper = styled.div`
    position: relative;
`;

const MyLocationWrapper = styled.div`
    position: absolute;
    bottom: 20px;
    z-index: 10;
    cursor: pointer;
    left: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;

    ${media.large`
        bottom: 20px;
    `}
`;

const StyledMyLocationIcon = styled(BiTargetLock)`
    color: #fff;
    font-size: 30px;

    ${media.large`
        font-size: 40px;
    `}
`;

const MyLocationText = styled.span`
    color: #fff;
    font-size: 13px;
`;

const PaginationWrapper = styled.div`
    position: absolute;
    bottom: 140px;
    left: 0;
    right: 0;
    margin: 0 auto;
    width: 570px;
    max-width: 98%;
    z-index: 11;
    text-align: center;
`;