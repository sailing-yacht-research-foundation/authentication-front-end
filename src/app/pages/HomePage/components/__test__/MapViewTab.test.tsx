import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { MapViewTab } from '../MapViewTab';

const shallowRenderer = createRenderer();

describe('<MapViewTab />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<MapViewTab />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
