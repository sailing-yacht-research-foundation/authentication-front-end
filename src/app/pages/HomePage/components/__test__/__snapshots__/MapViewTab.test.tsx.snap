// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapViewTab /> should render and match the snapshot 1`] = `
<React.Fragment>
  <InteractiveMap
    altitude={1.5}
    asyncRender={false}
    attributionControl={true}
    bearing={0}
    className=""
    clickRadius={0}
    container={<body />}
    disableTokenWarning={false}
    doubleClickZoom={true}
    dragPan={true}
    dragRotate={true}
    getCursor={[Function]}
    height="calc(100vh - 73px - 76px)"
    keyboard={true}
    latitude={37.8}
    longitude={-122.4}
    mapOptions={Object {}}
    mapStyle="mapbox://styles/jweisbaum89/cki2dpc9a2s7919o8jqyh1gss"
    mapboxApiAccessToken="no-token"
    mapboxApiUrl="https://api.mapbox.com"
    maxPitch={60}
    maxZoom={24}
    minPitch={0}
    minZoom={0}
    onClick={null}
    onContextMenu={[Function]}
    onError={[Function]}
    onHover={null}
    onLoad={[Function]}
    onNativeClick={null}
    onResize={[Function]}
    onStateChange={[Function]}
    onTransitionEnd={[Function]}
    onTransitionInterrupt={[Function]}
    onTransitionStart={[Function]}
    onViewStateChange={null}
    onViewportChange={[Function]}
    pitch={0}
    preserveDrawingBuffer={false}
    preventStyleDiffing={false}
    reuseMaps={false}
    scrollZoom={true}
    style={null}
    touchAction="none"
    touchRotate={false}
    touchZoom={true}
    transitionDuration={0}
    transitionEasing={[Function]}
    transitionInterpolator={
      LinearInterpolator {
        "around": undefined,
        "propNames": Array [
          "longitude",
          "latitude",
          "zoom",
          "bearing",
          "pitch",
        ],
      }
    }
    transitionInterruption={1}
    visibilityConstraints={
      Object {
        "maxPitch": 60,
        "maxZoom": 24,
        "minPitch": 0,
        "minZoom": 0,
      }
    }
    visible={true}
    width="100vw"
    zoom={14}
  />
  <styled.div>
    <styled.div>
      <Styled(Input)
        onChange={[Function]}
        placeholder="Search races with SYRF"
        type="search"
        value=""
      />
      <Styled(Component) />
    </styled.div>
    <styled.div>
      <a
        href="/"
        onClick={[Function]}
      >
        Advanced search?
      </a>
    </styled.div>
  </styled.div>
</React.Fragment>
`;
